src/chunk.c

```diff
@@ -2739,12 +2740,14 @@ ts_chunk_has_associated_compressed_chunk(int32 chunk_id)
 	ts_scanner_foreach(&iterator)
 	{
 		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
-		bool isnull;
-		heap_getattr(ti->tuple, Anum_chunk_compressed_chunk_id, ti->desc, &isnull);
-		compressed = !isnull; // isnull is false when compress chunk id is set
+		bool chunk_id_isnull, dropped_isnull;
+		Datum dropped;
+		heap_getattr(ti->tuple, Anum_chunk_compressed_chunk_id, ti->desc, &chunk_id_isnull);
```

Comment:
I `dropped=true`, shouldn't `compressed_chunk_id` be NULL? Or, is the compressed chunk's metadata also retained with `dropped=true`?

This whole thing is very ambiguous and I think we need to make this a lot more robust in general.

Reply:
Yes, but I think that is something we should focus on after the release.